recipes/dcmtk/all/conanfile.py

```diff
@@ -73,42 +70,36 @@ def configure(self):
             del self.options.fPIC
         if self.settings.os == "Windows":
             del self.options.with_tcpwrappers
-        
-        # Looking into source code, it appears that OpenJPEG and libsndfile are not used
-        if self.options.with_openjpeg != "deprecated":
-            self.output.warn("with_openjpeg option is deprecated, do not use anymore")
-        if self.options.with_libsndfile != "deprecated":
-            self.output.warn("with_libsndfile option is deprecated, do not use anymore")
+
+    def validate(self):
+        if hasattr(self, "settings_build") and tools.cross_building(self) and self.settings.arch == "armv8":
+            # FIXME: Probable issue with flags, build includes header 'mmintrin.h'
+            raise ConanInvalidConfiguration("Cross building to 'arm' is not supported (yet)")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def requirements(self):
         if self.options.charset_conversion == "libiconv":
             self.requires("libiconv/1.16")
         elif self.options.charset_conversion == "icu":
-            self.requires("icu/68.2")
+            self.requires("icu/69.1")
         if self.options.with_libxml2:
-            self.requires("libxml2/2.9.10")
+            self.requires("libxml2/2.9.12")
         if self.options.with_zlib:
             self.requires("zlib/1.2.11")
         if self.options.with_openssl:
-            # FIXME: Bump openssl. dcmtk build files have a logic to detect
-            #        various openssl API but for some reason it fails with 1.1 API
+            # Library uses opaque type 'ssl_ctx_st' from openssl. Not available in >=1.1.0.
+            #   and it fails to build only in Windows
             self.requires("openssl/1.0.2u")
```

Comment:
`openssl/1.1.1k` fails in Windows (see build for commit 8d42355)... we could use different versions for Windows and Linux/Mac, but  probably we don't want to do that, right?

Reply:
what is failing, exactly?

Reply:
You can check the full build output of that commit, but basically:

```
dcmtls\libsrc\tlslayer.cc(622): error C2027: use of undefined type 'ssl_ctx_st'
```

There is nothing you can include in the new version to get the definition of that type.